src/shapes/itext.class.js

```diff
@@ -276,31 +282,57 @@
     },
 
     /**
-     * Prepare and clean the contextTop
+     * Prepare top context
+     * @returns {CanvasRenderingContext2D|undefined} ctx
      */
-    clearContextTop: function(skipRestore) {
-      if (!this.isEditing || !this.canvas || !this.canvas.contextTop) {
+    prepareContextTop: function () {
+      if (!this.canvas || !this.canvas.contextTop) {
         return;
       }
       var ctx = this.canvas.contextTop, v = this.canvas.viewportTransform;
       ctx.save();
       ctx.transform(v[0], v[1], v[2], v[3], v[4], v[5]);
       this.transform(ctx);
-      this._clearTextArea(ctx);
-      skipRestore || ctx.restore();
+      return ctx;
+    },
+
+    /**
+     * Prepare and clear top context
+     * @returns {CanvasRenderingContext2D|undefined} ctx
+     */
+    _clearContextTop: function () {
+      var ctx = this.prepareContextTop();
+      ctx && this._clearTextArea(ctx);
+      return ctx;
+    },
+
+    /**
+     * clear top context
+     * @returns {CanvasRenderingContext2D|undefined} ctx
+     */
+    clearContextTop: function() {
+      if (!this.isEditing) {
+        return;
+      }
+      var ctx = this._clearContextTop();
+      ctx && ctx.restore();
+      return ctx;
```

Comment:
ok clearContextTop and  _clearContextTop  is worse than before imho.
Is not fixable now because everything is interleaved and we use it i 6 places.
2 of those 6 wants to execute code between the cleaning and the restore part.

Reply:
possibly in the future we want to have clearing more blunt and the places that need to execute extra code will have their own save/restore couple and transform

Reply:
What are you suggesting?
I worked around with what I found.

Reply:
I can merge `clearContextTop` and `_clearContextTop` into `clearContextTop(restore: boolean)`.
That's the only difference between the two.

Reply:
That's how it was kind of